src/oop_ext/interface/_tests/test_interface.py

```diff
@@ -1077,3 +1077,39 @@ def AfterCaption(*args):
 
     assert Foo.GetCaption() == "Foo"
     assert Foo().GetValues("m") == [0.1, 10.0]
+
+
+def testGenericInterface() -> None:
+    """Generic interfaces need to support checking on both generic and non-generic implementers"""
+    from typing import FrozenSet, Generic, TypeVar
+
+    T = TypeVar("T", covariant=True)
+
+    class IFoo(Interface, Generic[T], TypeCheckingSupport):
+        # Class instance defined here as a workaround for this class to work in Python 3.6.
+        __abstractmethods__: FrozenSet = frozenset()
```

Comment:
@nicoddemus 
Do you think we could drop python 3.6 support in oop-ext?

Reply:
Definitely, IMO we can do this for all of our OSS projects, supporting >=3.7 or even >=3.10 only, if the former is too much work.